File: 2015/01/01.py
FILENAME = "2015/01/input.txt"


def part_one():
    floor = 0

    with open(FILENAME, "r", encoding="UTF-8") as file:
        for line in file:
            for char in line:
                if char == "(":
                    floor += 1
                elif char == ")":
                    floor -= 1

    print(floor)


def part_two():
    floor = 0

    with open(FILENAME, "r", encoding="UTF-8") as file:
        for line in file:
            for index, char in enumerate(line):
                if char == "(":
                    floor += 1
                elif char == ")":
                    floor -= 1

                if floor == -1:
                    print(index + 1)
                    return


if __name__ == "__main__":
    part_one()
    part_two()

File: 2015/01/01.c
#include <stdio.h>

const char *FILE_NAME = "input.txt";

void part_one()
{
    FILE *file = fopen(FILE_NAME, "r");

    int floor = 0;
    int ch;
    while ((ch = fgetc(file)) != EOF)
    {
        if (ch == '(')
        {
            floor++;
        }
        else
        {
            floor--;
        }
    }

    fclose(file);

    printf("%d\n", floor);
}

void part_two()
{
    FILE *file = fopen(FILE_NAME, "r");

    int floor = 0;
    int index = 0;
    int ch;
    while ((ch = fgetc(file)) != EOF)
    {
        if (ch == '(')
        {
            floor++;
        }
        else
        {
            floor--;
        }

        if (floor == -1)
        {
            printf("%d\n", index + 1);
            break;
        }

        index++;
    }

    fclose(file);
}

int main()
{
    part_one();
    part_two();
    return 0;
}
